README.md

```diff
@@ -1,53 +1,153 @@
 
+### I.b. Objectives
 
-### Objectives
+The objective of our project is to design a classification model to accurately predict which final judgement label would most likely reflect the general agreement of the AITA community in response to a given scenario. In the current state of the subreddit, a winning verdict is determined after 18 hours. With our project, a verdict would be more quickly brought to the table for conflicted users to find out if they had been in the wrong, and to be sooner rid of their worries.
 
-The objective of our project is to design a classification model to accurately predict which final judgement label would most likely reflect the general agreement of the AITA community in response to a given scenario. In the current state of the subreddit, a winning verdict is determined after 18 hours. With our project, a verdict would be more quickly brought to the table for conflicted users to find out if they had be_en in the wrong, and to be sooner rid of their worries. 
+### I.c. Related Works
 
-## Related works
+> todo
 
 _Liang, H., Sun, X., Sun, Y. et al. J Wireless Com Network (2017) 2017: 211. https://doi.org/10.1186/s13638-017-0993-1_
 
-## Materials and Methods
+## II. Materials and Methods
+
+### II.a. Dataset
+
+[Pushshift](https://pushshift.io/) is a big data storage project that copies Reddit submissions and comments. Its goal is to provide a common API for anyone to access this data on a large scale. Using this, we will extract all the posts submitted to AITA including their labels.
+
+Each item in the resultant dataset will have the following attributes that we deemed to be useful for classfication:
+
+- Title - the title of the reddit post
+- Content - the content of the reddit post
+- Label - the concluded label associated to the post
+
+The data extracted from Pushshift dates from Monday, February 24, 2014 9:50:51 PM to Saturday, April 4, 2020 8:41:28 PM. The number of posts retrieved total to 37679 items.
+
+### II.b. Technologies and Algorithms
+
+The core of the project is written in Python and requires version 3.5+. The project has been run with version 3.7.5. Apache Spark is used for this project in parallelizing computations on the data.
+
+Natural language processing techniques are first used to clean up the data extracted from Pushshift. Code related to this process can be found under `/data_preparation`. Because the data consists of written texts from the title and content of the AITA posts, it is filtered and processed with normalization to enhance results by removing noise and keeping the core information only. The techniques used for this pre-processing are stopwords removal, digits removal, and lemmatization. We also chose to remove singular proper nouns (tagged as NNP by nltk) as many names are used by posters in telling their stories but do not provide pertinence for classificaiton. The cleaned data is prepared once and stored into a database on a server to be used by all following algorithms.
```

Comment:
There's a typo in the word 'classification' in the before-last sentence. Also, I would add that the removal of proper nouns is based on our assumption that they would not provide any benefit to the classification, instead of stating it as a fact.